node/src/reactor/joiner.rs

```diff
@@ -333,11 +329,16 @@ impl ReactorEvent for JoinerEvent {
             JoinerEvent::TrieRequestIncoming(_) => "TrieRequestIncoming",
             JoinerEvent::TrieResponseIncoming(_) => "TrieResponseIncoming",
             JoinerEvent::FinalitySignatureIncoming(_) => "FinalitySignatureIncoming",
-            JoinerEvent::ContractRuntimeRequest(_) => "ContractRuntimeRequest",
         }
     }
 }
 
+impl From<ContractRuntimeRequest> for JoinerEvent {
```

Comment:
This is essentially putting the routing logic ("where does the `ContractRuntimeRequest` go") into the `From<>` impl, instead of putting it in `dispatch_event`. 

This is different from how the rest of the system is supposed to do it. This `From<_>` impl should be a derived `#[from]` impl on the event, and the resulting `Event::ContractRuntimeRequest` be forwarded to the componet via `dispatch_event`.

We can change the pattern, but I want to discuss the effects of that first :)

Reply:
The change has been reverted, we'll discuss before further refactoring.